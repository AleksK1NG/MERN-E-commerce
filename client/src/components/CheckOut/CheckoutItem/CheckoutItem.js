import React from 'react'
import { connect } from 'react-redux'
import { addCartItem, clearItemFromCart, deleteCartItem } from '../../../storeModules/cart/cartActions'

import {
  CheckoutItemContainer,
  ImageContainer,
  QuantityContainer,
  RemoveButtonContainer,
  TextContainer
} from './CheckoutItem.styles'

export const CheckoutItem = ({ cartItem, deleteCartItem, addCartItem, clearItemFromCart }) => {
  const { imageUrl, name, quantity, price } = cartItem

  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <img src={imageUrl} alt="item" />
      </ImageContainer>
      <TextContainer>{name}</TextContainer>
      <QuantityContainer>
        <div onClick={() => deleteCartItem(cartItem)}>&#10094;</div>
        <span>{quantity}</span>
        <div onClick={() => addCartItem(cartItem)}>&#10095;</div>
      </QuantityContainer>
      <TextContainer>{price}</TextContainer>
      <RemoveButtonContainer onClick={() => clearItemFromCart(cartItem)}>&#10005;</RemoveButtonContainer>
    </CheckoutItemContainer>
  )
}

export default connect(
  null,
  { deleteCartItem, addCartItem, clearItemFromCart }
)(CheckoutItem)
