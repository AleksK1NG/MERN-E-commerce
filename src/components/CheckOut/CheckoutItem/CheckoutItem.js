import React from 'react'
import { connect } from 'react-redux'
import './CheckoutItem.styles.scss'
import { addCartItem, clearItemFromCart, deleteCartItem } from '../../../storeModules/cart/cartActions'

const CheckoutItem = ({ cartItem, deleteCartItem, addCartItem, clearItemFromCart }) => {
  const { imageUrl, name, quantity, price } = cartItem
  return (
    <div className="checkout-item">
      <div className="image-container">
        <img src={imageUrl} alt="item" />
      </div>
      <span className="name">{name}</span>
      <span className="quantity">
        <div className="arrow" onClick={() => deleteCartItem(cartItem)}>
          &#10094;
        </div>
        <span className="value">{quantity}</span>
        <div className="arrow" onClick={() => addCartItem(cartItem)}>
          &#10095;
        </div>
      </span>
      <span className="price">{price}</span>
      <div className="remove-button" onClick={() => clearItemFromCart(cartItem)}>
        &#10005;
      </div>
    </div>
  )
}

export default connect(
  null,
  { deleteCartItem, addCartItem, clearItemFromCart }
)(CheckoutItem)
