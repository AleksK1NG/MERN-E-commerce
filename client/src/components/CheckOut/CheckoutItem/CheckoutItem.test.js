import React from 'react'
import { shallow } from 'enzyme'
import { CheckoutItem } from './CheckoutItem'

describe('Test CheckoutItem component', () => {
  let wrapper
  let mockClearItem
  let mockAddItem
  let mockRemoveItem

  beforeEach(() => {
    mockClearItem = jest.fn()
    mockAddItem = jest.fn()
    mockRemoveItem = jest.fn()

    const mockProps = {
      cartItem: {
        imageUrl: 'www.testImage.com',
        price: 10,
        name: 'hats',
        quantity: 2
      },
      clearItemFromCart: mockClearItem,
      addCartItem: mockAddItem,
      deleteCartItem: mockRemoveItem
    }

    wrapper = shallow(<CheckoutItem {...mockProps} />)
  })

  it('should CheckoutItem component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call clearItem when remove button is clicked', () => {
    wrapper.find('RemoveButtonContainer').simulate('click')

    expect(mockClearItem).toHaveBeenCalled()
  })

  it('should call addCartItem when left arrow is clicked', () => {
    wrapper
      .find('QuantityContainer')
      .find('div')
      .at(1)
      .simulate('click')

    expect(mockAddItem).toHaveBeenCalled()
  })

  it('should call deleteCartItem when left arrow is clicked', () => {
    wrapper
      .find('QuantityContainer')
      .find('div')
      .at(0)
      .simulate('click')

    expect(mockRemoveItem).toHaveBeenCalled()
  })
})
