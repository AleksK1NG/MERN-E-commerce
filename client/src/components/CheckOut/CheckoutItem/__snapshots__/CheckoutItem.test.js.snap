// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckoutItem component should CheckoutItem component 1`] = `ShallowWrapper {}`;
